### PR TITLE
Remove raising an error if the stream_id is not sequentially increasing

### PR DESCRIPTION
**Pull request by TwistedTwigleg**
*Description of changes:*

Fixes an issue where a race condition when unsubscribing and publishing in Greengrass can lead to the unsubscribe arriving before the publish, causing the packets to be out of order and lead to a total disconnect. This PR will make it just log the error but otherwise continue.

_________

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/event_stream_rpc_server.c [hunk 1]
```diff
@@ -922,13 +922,6 @@ static void s_route_message_by_type(
                     AWS_LS_EVENT_STREAM_RPC_SERVER,
                     "id=%p: stream_id is invalid because it's not sequentially increasing",
                     (void *)connection);
-
-                aws_raise_error(AWS_ERROR_EVENT_STREAM_RPC_PROTOCOL_ERROR);
```
Comment (MikeDombo) [inline source/event_stream_rpc_server.c:926]: I believe this error is also rais-able by the client, so the same change should be applied there.

File: source/event_stream_rpc_server.c [hunk 2]
```diff
@@ -922,13 +922,6 @@ static void s_route_message_by_type(
...
-                s_send_connection_level_error(
```
Comment (MikeDombo) [inline source/event_stream_rpc_server.c:927]: I'm not sure about removing this actually. Should be OK to send the error, we just don't want to close the connection when this specific error is received by client/server

File: source/event_stream_rpc_server.c (hunk 1 above)
Comment (TwistedTwigleg) [reply source/event_stream_rpc_server.c:926]: The error is raise-able by the client, but the client doesn't have a condition for checking for stream_id non-sequentially increasing that I could see. I will double check though.

**Review by bretambrose — COMMENTED** [review]
I'm confused why we're removing this.  I don't understand how the race condition described would manifest itself as a stream id error.

**Pull request closed without merging**